core/src/state.rs

```diff
@@ -2015,15 +2004,41 @@ mod tests {
             RoleIdWithOwner::new(gen_account_in("wonderland").0, "4".parse().unwrap()),
             RoleIdWithOwner::new(gen_account_in("0").0, "5".parse().unwrap()),
             RoleIdWithOwner::new(gen_account_in("1").0, "6".parse().unwrap()),
-        ];
-        let map = BTreeSet::from(roles);
+        ]
+        .map(|role| (role, ()));
+        let map = Storage::from_iter(roles);
 
-        let range = map
+        let view = map.view();
+        let range = view
             .range(RoleIdByAccountBounds::new(&account_id))
             .collect::<Vec<_>>();
         assert_eq!(range.len(), 2);
-        for role in range {
+        for (role, ()) in range {
             assert_eq!(&role.account, &account_id);
         }
     }
+
+    #[test]
+    fn account_domain_range() {
+        let accounts = [
+            gen_account_in("wonderland").0,
+            gen_account_in("wonderland").0,
+            gen_account_in("a").0,
+            gen_account_in("b").0,
+            gen_account_in("z").0,
+            gen_account_in("z").0,
+        ]
+        .map(|account| (account, ()));
+        let map = Storage::from_iter(accounts);
+
+        let domain_id = "kingdom".parse().unwrap();
+        let view = map.view();
+        let range = view
+            .range(AccountByDomainBounds::new(&domain_id))
+            .collect::<Vec<_>>();
+        assert_eq!(range.len(), 0);
+        for (account, ()) in range {
+            assert_eq!(&account.domain, &domain_id);
+        }
```

Comment:
I think this iterates over an empty vec and asserts nothing

Reply:
oh, right, it's leftover from my testing

Reply:
fixed it to just count + added extra check